src/main/java/org/mockito/internal/creation/bytebuddy/InlineBytecodeGenerator.java

```diff
@@ -228,33 +228,52 @@ public synchronized void mockClassConstruction(Class<?> type) {
     private static void assureInitialization(Class<?> type) {
         try {
             Class.forName(type.getName(), true, type.getClassLoader());
-        } catch (Throwable ignore) {
+        } catch (ExceptionInInitializerError e) {
+            throw new MockitoException(
+                    join(
+                            "Cannot instrument class that could not be initialized.",
+                            "",
+                            "A class that is not initializable would always fail during instrumentation.",
+                            "Static initializers are never mocked by Mockito to avoid permanent disintegration of classes."),
+                    e.getException());
+        } catch (Throwable ignored) {
```

Comment:
Well, one never knows what JVMs do, especially if not HotSpot. Retaining the original exception is always a good idea.

Reply:
Ah sorry, I think you have misunderstood me. I meant that it might be good to remove this `catch` block which silently discards any `Throwable`, so the caller has to deal with them instead. Or only discard the relevant exceptions, i.e. `ClassNotFoundException` for hidden classes?
This `catch` block was likely also the reason why the previous error (which this pull request improves) was so cryptic / did not contain the relevant information.

Sorry in case my previous review comments were too intrusive.